print slowest tests

kain88-de opened the pull request:
enable test profiling again. This will show us the slowest tests like we had for nosetest.

pytest really comes with batteries included

kain88-de commented:
Output for test runtimes 

```
========================== slowest 50 test durations ===========================

92.81s call     MDAnalysisTests/coordinates/test_lammps.py::TestLAMMPSDATAWriter_data::test_Writer_atoms

78.88s call     MDAnalysisTests/analysis/test_gnm.py::TestGNM::test_closeContactGNMAnalysis

53.73s call     MDAnalysisTests/analysis/test_gnm.py::TestGNM::test_closeContactGNMAnalysis_weights_None

19.84s call     MDAnalysisTests/coordinates/test_pdb.py::TestMultiPDBReader::test_conect_bonds_all

19.24s call     MDAnalysisTests/coordinates/test_pdb.py::TestMultiPDBReader::test_conect_bonds_conect

17.69s call     MDAnalysisTests/coordinates/test_gro.py::TestGROLargeWriter::test_writer_large

12.37s call     MDAnalysisTests/coordinates/test_lammps.py::TestLAMMPSDATAWriter_data::test_Writer_dimensions

11.87s call     MDAnalysisTests/coordinates/test_lammps.py::TestLAMMPSDATAWriter_cnt::test_Writer_atoms

10.11s call     MDAnalysisTests/coordinates/test_pdb.py::TestMultiPDBReader::test_iteration

9.89s call     MDAnalysisTests/coordinates/test_pdb.py::TestMultiPDBReader::test_n_atoms_frame

9.83s call     MDAnalysisTests/coordinates/test_pdb.py::TestMultiPDBReader::test_rewind

9.72s call     MDAnalysisTests/coordinates/test_gro.py::TestGROLargeWriter::test_write_trajectory_atomgroup

9.71s call     MDAnalysisTests/coordinates/test_pdb.py::TestMultiPDBReader::test_numconnections

9.71s call     MDAnalysisTests/coordinates/test_pdb.py::TestMultiPDBReader::test_n_frames

9.70s call     MDAnalysisTests/coordinates/test_pdb.py::TestMultiPDBReader::test_slice_iteration

9.40s call     MDAnalysisTests/coordinates/test_gro.py::TestGROLargeWriter::test_write_trajectory_universe

8.31s call     MDAnalysisTests/analysis/test_encore.py::TestEncore::test_ces_error_estimation_ensemble_bootstrap

8.02s setup    MDAnalysisTests/utils/test_altloc.py::test_bonds

8.00s setup    MDAnalysisTests/utils/test_altloc.py::test_write_read

7.96s setup    MDAnalysisTests/utils/test_altloc.py::test_atomgroups

7.72s call     MDAnalysisTests/coordinates/test_gro.py::TestGROLargeWriter::test_writer_large_residue_count

7.11s call     MDAnalysisTests/analysis/test_hbonds.py::TestHydrogenBondAnalysis::test_true_traj

6.99s call     MDAnalysisTests/analysis/test_hbonds.py::TestHydrogenBondAnalysisHeuristic::test_true_traj

6.96s call     MDAnalysisTests/analysis/test_hbonds.py::TestHydrogenBondAnalysisHeavy::test_true_traj

6.94s call     MDAnalysisTests/analysis/test_hbonds.py::TestHydrogenBondAnalysisHeavyFail::test_true_traj

6.75s call     MDAnalysisTests/analysis/test_encore.py::TestEncore::test_hes_custom_weights

6.01s call     MDAnalysisTests/core/test_atomselections.py::TestSelectionsXTC::test_same_fragment

5.89s call     MDAnalysisTests/analysis/test_pca.py::TestPCA::test_pcs

5.81s call     MDAnalysisTests/formats/test_libdcd.py::test_written_remarks_property

5.60s call     MDAnalysisTests/analysis/test_encore.py::TestEncore::test_hes_error_estimation

5.13s call     MDAnalysisTests/analysis/test_pca.py::TestPCA::test_cum_var

4.90s call     MDAnalysisTests/analysis/test_pca.py::TestPCA::test_transform_mismatch

4.60s call     MDAnalysisTests/analysis/test_pca.py::TestPCA::test_cov

4.60s call     MDAnalysisTests/coordinates/test_gro.py::TestGROLargeWriter::test_write_selection

4.55s call     MDAnalysisTests/analysis/test_pca.py::TestPCA::test_different_steps

4.46s call     MDAnalysisTests/core/test_topologyobjects.py::TestTopologyGroup::test_TG_strict_intersection

4.34s call     MDAnalysisTests/coordinates/test_pdb.py::TestMultiPDBWriter::test_write_atoms

4.28s call     MDAnalysisTests/analysis/test_hole.py::TestHoleModule::test_hole_module_fd_closure

4.09s call     MDAnalysisTests/coordinates/test_gro.py::TestGROLargeWriter::test_write_trajectory_timestep

4.05s call     MDAnalysisTests/analysis/test_gnm.py::TestGNM::test_generate_kirchoff

4.02s call     MDAnalysisTests/analysis/test_encore.py::TestEncore::test_dres_error_estimation

3.70s call     MDAnalysisTests/analysis/test_encore.py::TestEncore::test_ces_error_estimation

3.68s call     MDAnalysisTests/analysis/test_pca.py::TestPCA::test_cosine_content

3.56s call     MDAnalysisTests/coordinates/test_lammps.py::TestLAMMPSDATAWriter_data_partial::test_Writer_dimensions

3.51s call     MDAnalysisTests/analysis/test_density.py::TestNotWithin::test_not_within

3.49s call     MDAnalysisTests/coordinates/test_lammps.py::TestLAMMPSDATAWriter_data_partial::test_Writer_atoms

3.34s call     MDAnalysisTests/analysis/test_density.py::Test_density_from_Universe::test_density_from_Universe_update_selection

3.29s call     MDAnalysisTests/analysis/test_psa.py::TestPSAnalysis::test_dist_mat_to_vec_input_numpy_integer_32

3.27s call     MDAnalysisTests/analysis/test_psa.py::TestPSAnalysis::test_dist_mat_to_vec_i_less_j

3.22s call     MDAnalysisTests/analysis/test_psa.py::TestPSAnalysis::test_reversal_hausdorff
```

orbeckst commented on .travis.yml:
```diff
@@ -24,15 +24,15 @@ env:
     - SETUP_CMD=""
     - BUILD_CMD="pip install -v package/ && pip install testsuite/"
-    - CONDA_DEPENDENCIES="mmtf-python nose=1.3.7 mock six biopython networkx cython joblib nose-timer matplotlib scipy griddataformats pytest-cov pytest-xdist hypothesis"
-    - CONDA_ALL_DEPENDENCIES="mmtf-python nose=1.3.7 mock six biopython networkx cython joblib nose-timer matplotlib netcdf4 scikit-learn scipy griddataformats seaborn coveralls clustalw=2.1 pytest-cov pytest-xdist hypothesis"
```
Why do you not need to install the plugins anymore?

And if they are really part of `pytest` then you can also remove them from the `testsuite/setup.py` but please check first in a fresh venv that `pip install pytest` is sufficient.

kain88-de replied:
CI helpers installs those for us. They don't need to be in the setup.py since they just change calling options for pytest.

orbeckst commented on .travis.yml:
```diff
@@ -24,15 +24,15 @@ env:
     - NOSE_FLAGS="--processes=2 --process-timeout=400 --no-open-files --with-timer --timer-top-n 50"
     - NOSE_TEST_LIST=""
-    - PYTEST_FLAGS="--disable-pytest-warnings -n 2"
+    - PYTEST_FLAGS="--disable-pytest-warnings --numprocesses 2 --durations=50"
```
Is this all that's needed to enable timing information?! Nice...

orbeckst commented:
So if understand you correctly, https://github.com/MDAnalysis/mdanalysis/blob/develop/testsuite/setup.py remains unchanged and continued to list xdist and cob as dependencies. Ok. 

--
Oliver Beckstein
email: orbeckst@gmail.com


> Am Jul 14, 2017 um 09:11 schrieb Max Linke <notifications@github.com>:
> 
> @kain88-de commented on this pull request.
> 
> In .travis.yml:
> 
> >      - PYTEST_LIST="testsuite/MDAnalysisTests/lib testsuite/MDAnalysisTests/formats testsuite/MDAnalysisTests/coordinates testsuite/MDAnalysisTests/utils testsuite/MDAnalysisTests/topology testsuite/MDAnalysisTests/auxiliary testsuite/MDAnalysisTests/core testsuite/MDAnalysisTests/analysis"
>      - NOSE_COVERAGE_FILE="nose_coverage"
>      - PYTEST_COVERAGE_FILE="pytest_coverage"
>      - MAIN_CMD="pytest ${PYTEST_LIST} ${PYTEST_FLAGS}"
>      - SETUP_CMD=""
>      - BUILD_CMD="pip install -v package/ && pip install testsuite/"
> -    - CONDA_DEPENDENCIES="mmtf-python nose=1.3.7 mock six biopython networkx cython joblib nose-timer matplotlib scipy griddataformats pytest-cov pytest-xdist hypothesis"
> -    - CONDA_ALL_DEPENDENCIES="mmtf-python nose=1.3.7 mock six biopython networkx cython joblib nose-timer matplotlib netcdf4 scikit-learn scipy griddataformats seaborn coveralls clustalw=2.1 pytest-cov pytest-xdist hypothesis"
> CI helpers installs those for us. They don't need to be in the setup.py since they just change calling options for pytest.
> 
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub, or mute the thread.
>

kain88-de commented:
@orbeckst I removed the `pytest-xdist` and `pytest-cov` package from the requirements. The tests can be run without these packages. I would suggest adding to developer information later on.

I also removed the `pytest.raises` package and converted all test that use it to the `pytest.raises` context manager we agreed on in the style guide lines. 

Also the numpy dependency doesn't need to be here since it's already fulfilled in MDAnalysis

kain88-de commented:
@richardjgowers feel free to merge if you like the changes
